tools/nginz_disco/nginz_disco.sh

```diff
@@ -40,16 +41,19 @@ function valid_ipv6() {
 function upstream() {
     name=$1
     port=${2:-'8080'}
-    ips=$(dig +short +retries=3 +search ${name} | sort)
+    ips=$(dig +short +retries=3 +search "${name}" | sort)
     unset servers
-    for ip in ${ips[@]}; do
-        if valid_ipv4 $ip || valid_ipv6 $ip; then
+    for ip in "${ips[@]}"; do
```

Comment:
**issue:** this is not equivalent. Test:
```
ips=$(dig +short +retries=3 +search "cloudflare.com" | sort) # ips has 2 elements
for ip in "${ips[@]}"; do echo $ip; done
```

this prints `104.16.132.229 104.16.133.229` in one line instead of two lines.